APISpec.md

```diff
@@ -0,0 +1,508 @@
+# Todo API Spec
+
+## API Resources
+
+* **User** - A User represents an owner of Todos and corresponds one-to-many to a Todo.
+
+* **Todo** - A Todo represents a task to be completed and corresponds many-to-one to a User.
+
+* **Session** - A Session represents an authenticated user session and corresponds many-to-one to a User.
```

Comment:
I think when I said to think of login as creating a session I explained it badly.

Having a full CRUD API for sessions and auth things is usually a very bad idea unless you are building some sort of auditing system that allows administrators to track user activity.

You can have just a regular `login` route only, my point was that if you are thinking in the REST mindset `login` isn't "logging in" it is "creating a user session resource"